src/path.js

```diff
@@ -19,15 +19,9 @@ var _curry2 = require('./internal/_curry2');
  *      R.path(['a', 'b'], {a: {b: 2}}); //=> 2
  *      R.path(['a', 'b'], {c: {b: 2}}); //=> undefined
  */
-module.exports = _curry2(function path(paths, obj) {
-  var val = obj;
-  var idx = 0;
-  while (idx < paths.length) {
-    if (val == null) {
-      return;
+module.exports = _curry2(function path(paths, obj, i) {
```

Comment:
Yeah definitely, one way to remove it would be to nest the function inside an other one calling for this one: you'd have the path(paths, obj) function calling inside it the recursiveFunc(paths, obj, 0).